consensus/src/epoch_manager.rs

```diff
@@ -778,6 +784,8 @@ impl EpochManager {
     }
 
     async fn start_new_epoch(&mut self, payload: OnChainConfigPayload) {
+        // The recovery manager exits before starting a new epoch. So, we unset the recovery_mode flag here.
+        self.recovery_mode = false;
```

Comment:
this is not what I meant, I was suggesting that in the branch below LivenessStorageData::FullRecoveryData we set recovery_mode = false and PartialRecoveryData we set it to true.